src/conn.c

```diff
@@ -1867,11 +1847,24 @@ static int _conn_connect(xmpp_conn_t *conn,
     if (!conn->domain)
         return XMPP_EMEM;
 
-    conn->sock = sock_connect(conn, host, port);
-    strophe_debug(conn->ctx, "xmpp", "sock_connect() to %s:%u returned %d",
-                  host, port, conn->sock);
-    if (conn->sock == -1)
-        return XMPP_EINT;
+    do {
+        conn->sock = sock_connect(conn->xsock);
+        if (conn->sock == -1)
+            return XMPP_EINT;
+
+        if (conn->sockopt_cb) {
+            /* Don't allow user to overwrite sockfd value. */
+            sock_t sock_copy = conn->sock;
+            rc = conn->sockopt_cb(conn, &sock_copy);
```

Comment:
Moving the invocation of `sockopt_cb()` out of `sock_connect()` means user can no longer use this callback to set any socket options before `connect()`, was it necessary to do this?

Reply:
Good point, I completely missed this. The idea was to remove xmpp_conn_t knowledge from the sock component. I will return old implementation back for now.